AlphaWallet/Settings/ViewModels/EditCustomRPCViewModel.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  EditCustomRPCViewModel.swift
+//  AlphaWallet
+//
+//  Created by Jerome Chan on 7/11/21.
+//
+
+import Foundation
+
+enum EditCustomRPCErrors: Error {
+    case unknown
+    case list([EditCustomRPCErrors]) // This is just used to hold the errors in a Result<>.
+    case chainNameInvalidField, rpcEndPointInvalidField, chainIDInvalidField, symbolInvalidField, explorerEndpointInvalidField, chainIDDuplicateField
+}
+
+struct EditCustomRPCViewModel {
+    
+    private let model: CustomRPC
+    
+    init(model: CustomRPC) {
+        self.model = model
+    }
+    
+    var chainID: String {
+        if model.chainID == 0 {
+            return R.string.localizable.chainID()
+        }
+        return String(model.chainID)
+    }
+    
+    var nativeCryptoTokenName: String {
+        return model.nativeCryptoTokenName ?? ""
+    }
+    
+    var chainName: String {
+        if model.chainName.isEmpty {
+            return R.string.localizable.addrpcServerNetworkNameTitle()
+        }
+        return model.chainName
+    }
+    
+    var symbol: String {
+        return model.symbol ?? R.string.localizable.symbol()
+    }
+    
+    var rpcEndPoint: String {
+        if !model.rpcEndpoint.isEmpty,
+           let _ = URL(string: model.rpcEndpoint) {
+            return model.rpcEndpoint
+        } else {
+            return R.string.localizable.addrpcServerRpcUrlTitle()
+        }
+    }
+    
+    var explorerEndpoint: String {
+        if let eep = model.explorerEndpoint,
+           !eep.isEmpty,
+           let _ = URL(string: eep) {
+            return eep
+        } else {
+            return R.string.localizable.addrpcServerBlockExplorerUrlTitle()
+        }
+    }
+    
+    var etherscanCompatibleType: String {
```

Comment:
This looks like it's unused

Reply:
So the view model should only have computed properties for model properties which are used in the view? Anything not used should not be implemented?

Reply:
Generally yes, if code is not used we should not add it, or remove it. If it's in new code, it could imply it's implemented and meant to be used but somehow left out (i.e it's a bug).